domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/stoptimes/NotTimepointStoptime.java

```diff
@@ -0,0 +1,36 @@
+package org.mobilitydata.gtfsvalidator.domain.entity.stoptimes;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class NotTimepointStoptime extends TimepointBase {
+
+    public NotTimepointStoptime(@NotNull String tripId,
+                                @Nullable String arrivalTime,
+                                @Nullable String departureTime,
+                                @NotNull String stopId,
+                                int stopSequence,
+                                @Nullable String stopHeadsign,
+                                @NotNull PickupType pickupType,
+                                @NotNull DropOffType dropOffType,
+                                @Nullable Float shapeDistTraveled) {
+        super(tripId, arrivalTime, departureTime, stopId, stopSequence, stopHeadsign, pickupType, dropOffType,
+                shapeDistTraveled);
+    }
+
+    public static class NotTimepointStoptimeBuilder extends TimepointBaseBuilder {
+
+        public NotTimepointStoptimeBuilder(@NotNull String tripId,
```

Comment:
As we discovered together yesterday, @NotNull annotation doesn't provoke a NullPointerException at runtime. Please add a null check and `throw new InvalidArgumentException` where applicable in **Builder** code.